### PR TITLE
Cobra cli

### PR DESCRIPTION
**Pull request by YujiOshima**
Introduce `spf13/cobra` to katib-cli.
CLI schema is drastically changed.
```
katib-cli [subcomand] [resource name] [options]
```
* [subcommand]
- create
- get
- stop
- push

* [resource name]
- study
- model

/assign @gaocegege

### CODE REVIEW

**Comment by YujiOshima** [comment]
Fix #54

File: cmd/cli/create.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+)
+
+//NewCommandGet generate run cmd
```
Comment (gaocegege) [inline cmd/cli/create.go]: The function name is `NewCommandCreate`. Maybe there is a typo

File: cmd/cli/push.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+)
+
+//NewCommandGet generate run cmd
```
Comment (gaocegege) [inline cmd/cli/push.go]: Same problem above.

File: cmd/cli/get-model.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+		r, err := c.GetStudies(context.Background(), req)
+		if err != nil {
+			log.Fatalf("GetStudy failed: %v", err)
+			return
+		}
+		if len(r.StudyInfos) > 0 {
+			var sInfo []*api.StudyInfo
+			for _, si := range r.StudyInfos {
+				if len(opt.args) > 0 {
+					if utf8.RuneCountInString(opt.args[0]) >= 7 {
```
Comment (gaocegege) [inline cmd/cli/get-model.go]: Could we add some comments here? I can not understand the code well.

File: cmd/cli/get.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		Long:  `list of resorces comannd can display includes: studies, study, trials, trial, models, model`,
+	}
+
+	//set local flag
+
+	//add subcommand
+	cmd.AddCommand(NewCommandGetStudies())
+	cmd.AddCommand(NewCommandGetStudy())
+	//	cmd.AddCommand(NewCommandGetTrials())
+	//	cmd.AddCommand(NewCommandGetTrial())
```
Comment (gaocegege) [inline cmd/cli/get.go]: I am wondering if we could enable the two command. WDYT

File: cmd/cli/katibctl.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+//Entry point
+func main() {
+	//init command
+	katibctl := NewRootCommand()
+	if err := katibctl.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
```
Comment (gaocegege) [inline cmd/cli/katibctl.go]: Please remove the line here.

File: cmd/cli/push-model.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	"google.golang.org/grpc"
+	yaml "gopkg.in/yaml.v2"
+)
+
+type pushModelOpt struct {
+	conf string
+	args []string
+}
+
+//NewCommandGetStudy generate run cmd
```
Comment (gaocegege) [inline cmd/cli/push-model.go]: Please update or remove the comment here.

File: cmd/cli/get.go (hunk 4 above)
Comment (YujiOshima) [reply cmd/cli/get.go]: Unfortunately, we don't have the get trial API in katib-manager.
I will refactor API and add low-level APIs (related #66 ). Then I will enable these commands.

Comment (gaocegege) [reply cmd/cli/get.go]: SGTM, I misunderstand the comments.

**Comment by gaocegege** [comment]
/lgtm

:tada:

**Pull request merged**